packages/serialization/json/src/jsonParseNode.ts

```diff
@@ -51,23 +50,46 @@ export class JsonParseNode implements ParseNode {
       }
     });
   };
+
   public getCollectionOfObjectValues = <T extends Parsable>(
-    type: ParsableFactory<T>
+    method: ParsableFactory<T>
   ): T[] | undefined => {
     return (this._jsonNode as unknown[])
       .map((x) => new JsonParseNode(x))
-      .map((x) => x.getObjectValue<T>(type));
+      .map((x) => x.getObjectValue<T>(method));
   };
-  public getObjectValue = <T extends Parsable>(type: ParsableFactory<T>): T => {
-    const result = type(this);
+
+  public getObjectValue = <T extends Parsable>(
+    parsableFactory: ParsableFactory<T>,
+    value: T = {} as T
```

Comment:
Since the parsable factory returns the deserializer and not an instance, we need to pass a value which is  a model object exclusively which can be used to set the result.

Reply:
@koros the prototype of this method still needs to be updated here